migrations/manual/educandu-2021-12-29-01-repair-image-tiles.js

```diff
@@ -0,0 +1,62 @@
+import { promises as fs } from 'fs';
+
+// eslint-disable-next-line camelcase
+export default class Educandu_2021_12_29_01_repair_image_tiles {
+  constructor(db) {
+    this.db = db;
+  }
+
+  async fixTile(documentLinkInfo) {
+    const revisions = await this.db.collection('documentRevisions').find({
+      'key': documentLinkInfo.documentKey,
+      'sections.key': documentLinkInfo.sectionKey,
+      'sections.revision': documentLinkInfo.sectionRevision
+    }).toArray();
+
+    for (const revision of revisions) {
+      const imageTilesSections = revision.sections.filter(section => section.type === 'image-tiles' && Array.isArray(section.content?.tiles))
+        .filter(section => section.key === documentLinkInfo.sectionKey && section.revision === documentLinkInfo.sectionRevision);
+
+      if (imageTilesSections.length !== 1) {
+        console.log(documentLinkInfo);
+        console.log('Found unexpected number of tile sections', imageTilesSections.length);
+      }
+
+      if (imageTilesSections[0].content.tiles.length < documentLinkInfo.tileIndex + 1) {
+        console.log(documentLinkInfo);
+        console.log('Number of tile sections', imageTilesSections[0].content.tiles.length, 'tile index: ', documentLinkInfo.tileIndex, 'revision', revision._id);
+      }
+      // eslint-disable-next-line no-await-in-loop
+      const doc = await this.db.collection('documents').findOne({ slug: documentLinkInfo.url });
+
+      if (imageTilesSections[0].content.tiles[documentLinkInfo.tileIndex]?.link.type !== 'internal') {
+        console.log(`Document info tile is not of expected type, but of type ${imageTilesSections[0].content.tiles[documentLinkInfo.tileIndex].link.type}`);
```

Comment:
Maybe better to extract `imageTilesSections[0].content.tiles[documentLinkInfo.tileIndex]?.link.type` in a variable, because in the second usage on line 33 the optional chaining operator is missing, but the `if` block could be entered with the condition evaluating to `if  (null !== 'internal')`

Reply:
Done, thanks! I also extracted some of the other long names to a variable for better readability.